compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -1139,26 +1138,24 @@ impl<'a> Linker for EmLinker<'a> {
     }
 
     fn export_symbols(&mut self, _tmpdir: &Path, _crate_type: CrateType, symbols: &[String]) {
+        use std::fmt::Write;
+
         debug!("EXPORTED SYMBOLS:");
 
+        let mut encoded = "[".to_string();
+        let mut symbols = symbols.iter();
+        if let Some(first_symbol) = symbols.next() {
+            write!(encoded, "{:?}", first_symbol).unwrap();
+        }
+        for symbol in symbols {
+            write!(encoded, ",{:?}", symbol).unwrap();
+        }
+        encoded.push(']');
+        debug!("{}", encoded);
+
         self.cmd.arg("-s");
 
         let mut arg = OsString::from("EXPORTED_FUNCTIONS=");
-        let mut encoded = String::new();
-
-        {
-            let mut encoder = json::Encoder::new(&mut encoded);
-            let res = encoder.emit_seq(symbols.len(), |encoder| {
-                for (i, sym) in symbols.iter().enumerate() {
-                    encoder.emit_seq_elt(i, |encoder| encoder.emit_str(&("_".to_owned() + sym)))?;
```

Comment:
It looks like the leading "_" is no longer included in your custom serialization code. Are there any tests verifying that the `EXPORTED_FUNCTIONS` we construct actually works?

Reply:
Right, emscripten is a tier 2 platform so it doesn't get any testing.

Reply:
Fixed